src/nvim/ui_compositor.c

```diff
@@ -101,6 +101,34 @@ bool ui_comp_should_draw(void)
   return composed_uis != 0 && valid_screen;
 }
 
+// Adjusts the position of a layer in the layers array based on its zindex.
+// Moves the layer either forward (if zindex increases) or backward (if zindex decreases).
+//
+// @param[in]  idx       The index of the current layer in the items array.
+// @param[in]  forward   True move forward otherwise move backward.
+void ui_comp_layers_adjust(size_t idx, bool forward)
```

Comment:
should we use the same language as the existing `ui_comp_raise_grid` function? Instead of referring to forward/backward, refer to higher/lower (or "raise").

Also why does the docstring say `if zindex increases`? The caller makes that decision, not this function.

```suggestion
// Raises or lowers the layer.
//
// @param[in]  layer_idx Index of the layer to be raised/lowered.
// @param[in]  raise     Raise the layer if true, else lower it.
void ui_comp_raise_layer(size_t layer_idx, bool raise)
```